sys/riscv/riscv/pmap.c

```diff
@@ -4237,6 +4265,65 @@ pmap_clear_modify(vm_page_t m)
 	rw_runlock(&pvh_global_lock);
 }
 
+#if __has_feature(capabilities)
+int
+pmap_emulate_capdirty(pmap_t pmap, vm_offset_t addr)
+{
```

Comment:
It would be nice to share this code with pmap_fault_fixup() if possible, perhaps by using `VM_PROT_CAP_WRITE` as the `ftype` passed to pmap_fault_fixup().